W-17856636-VPC-Upgrade-Changes (#777)
mulesoft/docs-runtime-manager: cloudhub/modules/ROOT/pages/vpc-upgrade.adoc

Limitation for Restricting Outbound Egress Traffic::
CloudHub VPC doesn't restrict outbound egress traffic, so removing the default firewall rule that allows this traffic can cause failures in your migrated private space.

== Understand Environment Association

> When upgrading your CloudHub VPC to a CloudHub 2.0 private space, the environments associated with your VPC stays in sync with the environments associated with the upgraded CloudHub 2.0 private space to avoid configuration drift.

Reviewer: 1. to avoid confusion, can we explicitly say, `When Upgrading & while upgrade is in progress i.e. CloudHub VPC & CloudHub 2.0 Private Space coexist...`
2. Can we also mention a point that if user associates bulk association on their CloudHub 2.0 Private Space, to avoid drift, tool will appropriately associate individual associations to the CloudHub VPC counterpart to avoid drift

Reviewer 2: For 1, we could say something like `During the upgrade process from CloudHub VPC to CloudHub 2.0 Private Space, the environments associated with your VPC stays in sync with the environments associated with the upgraded CloudHub 2.0 private space to avoid configuration drift.`
For 2, we should add this line `Additionally doing bulk environment associations for an upgraded Private Space can result in all environments in the same region to get associated with the underlying VPC. If those environments were previously associated with other CloudHub 1.0 VPCs, they will be disassociated.`

Reviewer: On 2 better would be:
```
Additionally doing bulk environment associations for an upgraded Private Space will result in environments getting associated with the underlying VPC. If those environments were previously associated with other CloudHub 1.0 VPCs in the same region, they will be disassociated.
```

Limitation for Restricting Outbound Egress Traffic::
CloudHub VPC doesn't restrict outbound egress traffic, so removing the default firewall rule that allows this traffic can cause failures in your migrated private space.

== Understand Environment Association

> During the upgrade process from CloudHub VPC to CloudHub 2.0 Private Space, the environments associated with your VPC stays in sync with the environments associated with the upgraded CloudHub 2.0 private space to avoid configuration drift.

Reviewer 3: ```suggestion
During the upgrade process from CloudHub VPC to CloudHub 2.0 Private Space, the environments associated with your VPC stay in sync with the environments associated with the upgraded CloudHub 2.0 Private Space to avoid configuration drift.
```

Environment association works differently in xref:cloudhub-2::ps-config-env.adoc[CloudHub 2.0] and xref:vpc-tutorial.adoc#update-an-existing-anypoint-vpc[CloudHub 1.0]. In CloudHub 2.0, an environment can be associated with multiple private spaces, whereas, in CloudHub, one environment can only be associated with one VPC in a given region.

Additionally, doing bulk environment association for an upgraded private space can result in all environments in the same region to get associated with the underlying VPC. If those environments were previously associated with other CloudHub 1.0 VPCs, they get disassociated.

[WARNING]

> The upgrade tooling doesn't support the upgrade of a default VPC. Uncheck the default VPC configuration before you can proceed to upgrade.

Reviewer 3: ```suggestion
The upgrade tooling doesn't support the upgrade of a default VPC. To proceed with the upgrade, disable the default VPC configuration.
```

Limitation for Restricting Outbound Egress Traffic::
CloudHub VPC doesn't restrict outbound egress traffic, so removing the default firewall rule that allows this traffic can cause failures in your migrated private space.

== Understand Environment Association

When upgrading your CloudHub VPC to a CloudHub 2.0 private space, the environments associated with your VPC stays in sync with the environments associated with the upgraded CloudHub 2.0 private space to avoid configuration drift.

> Environment association works differently in xref:cloudhub-2::ps-config-env.adoc[CloudHub 2.0] and xref:vpc-tutorial.adoc#update-an-existing-anypoint-vpc[CloudHub 1.0]. In CloudHub 2.0, an environment can be associated with multiple: whereas iin CloudHub, an environment can be associated to only one VPC.

Reviewer 2: In CloudHub 2.0, one environment can be associated with multiple private spaces, whereas in Cloudhub 1.0, one environment can only be associated with one VPC in a given region.

> === Upgrading the Default VPC

Reviewer 2: Remove this entire section, as we are not allowing upgrading default VPC. We can go with the `Warning` section below.
